README.md

```diff
@@ -65,70 +65,79 @@ MyClass classAnnotationsDo:
+### Annotation instantiation
+There is no special way how to instantiate annotation instances. It is up to your domain. An annotation is a plain object and can be stateful. You can add any domain specific variables to your annotations and add constructors to initialize them in the declarating methods. We give an example here after:
 
-The base internal state of annotation is initialized during registry creation.  Users should not think about it. 
+The internal state of annotation is initialized during registry creation.  Users should not think about it. 
 
-## Annotating Annotations
+### Annotating Annotations
 Annotations are just normal classes without any restrictions. You can also attach annotations to annotations like in other languages.
 
-## Forbidden annotation
-Annotation can forbid annotating of particular classes. For example it can forbid abstract classes.
+### Forbidden annotation
+Annotations can forbid annotating of particular classes. For example it can forbid abstract classes.
 ```Smalltalk
 MySpecificAnnotation >> isForbidden
       ^annotatedClass isAbstract
 ```
 Such annotations will not be added to the registry and forbidden classes will not include them.
 
 ## Contextual annotations
-Any annotation can be defined with context where it can be used. 
+Any annotation can be defined with a context where it can be used. 
 Imaging that you want annotate command classes with shortcuts for specific widgets:
+
 ```Smalltalk
-MyCommand class>>widget1ShortcutAnnotation
+MyCommand class >> widget1ShortcutAnnotation
    <classAnnotation>
    ^ShortcutAnnotation using: $r meta for: MyWidgetClass1
 
-MyCommand class>> widget2ShortcutAnnotation
+MyCommand class >> widget2ShortcutAnnotation
    <classAnnotation>
    ^ShortcutAnnotation using: $t meta for: MyWidgetClass2
 ```
-Then you can query all shortcuts which should be active for concrete widget: 
+It means that the same command MyCommand is bound the shortcut r in MyWidgetClass1 and the shortcut t in MyWidgetClass2.
+
+Then you can query all shortcuts which should be active for a concrete widget: 
 ```Smalltalk
 ShortcutAnnotation activeAnnotationsInContext: aMyWidgetInstance
 ShortcutAnnotation activeAnnotationsInContext: aMyWidgetInstance do: aBlock
 ```
-Context describes annotation users. And for contextual annotation lookup an instance of user should be provider. In that example it is aMyWidgetInstance.
 
-Declaring context using classes is simplest way how to restrict users which see annotation. By default given class is converted to SimpleAnnotationContext instance using #asAnnotationContext message. Then during annotation query it simply asks #isKindOf: for given user instance. 
+@@I do not get how this is working@@
+
+Context describes annotation users @@I do not get this sentence@@. And for contextual annotation lookup an instance of user should be provided. In that example it is aMyWidgetInstance.
```

Comment:
In shortcut example the annotation user is an instance of widget. We query annotations in context of their users:
```Smalltalk
ShortcutAnnotation activeAnnotationsInContext: aMyWidgetInstance
```
When we declare annotations we need to specify for what kind of users they should be visible. This "what kind of users" I call an annotation context. Annotation context implements method #describes: which in trivial form really checks the kind of user:
```Smalltalk
SimpleAnnotationContext>>describes: anAnnotationUser
  ^anAnnotationUser isKindOf: annotationUserClass
```
(if you look at this method in system it has extra indirection but it is not important here).

I thought many times what different terminology can be used here. Probably avoiding context name would be nice. I found that #condition is a good replacement. So it would be conditional annotations instead of contextual annotations (activeCondition instead of activeContext).
But I did not find good names for query methods:
- activeInstancesInContext: anAnnotationUser
- activeInstancesFor: aClass inContext: anAnnotationUser do: aBlock
- look at ClassAnnotation class side.